src/main/java/cubicchunks/CommonEventHandler.java

```diff
@@ -132,24 +142,52 @@ public void onPlayerJoinWorld(EntityJoinWorldEvent evt) {
 
     public static class ClientProxy extends ServerProxy {
 
+        private final List<Class<?>> allowedClientWorldClasses = ImmutableList.copyOf(new Class[] {
+                WorldClient.class,
+                WorldServer.class,
+                WorldServerMulti.class
+        });
+        
+        private final List<Class<?>> allowedClientChunkProviderClasses = ImmutableList.copyOf(new Class[] {
+                ChunkProviderClient.class,
+                ChunkProviderEnd.class,
+                ChunkProviderHell.class,
+                ChunkProviderFlat.class,
```

Comment:
WorldGenerators dont count as chunk prividers. Also make it `List<Class<? extends IChunkProvider>>`